tools/installer/installer.py

```diff
@@ -0,0 +1,940 @@
+        checksum_url = CHECKSUM_URL_BASE + {self.interviewer.chain} + "/latest/md5sum.txt"
+        # Get checksum file
+        published_checksum = self.exec(f"curl -s {checksum_url} | tail -1 | cut -d' ' -f 1").stdout.strip()
+        self.log(f"Comparing published checksum with local checksum")
+        local_checksum = self.exec(f"md5sum {file_path} | tail -1 | cut -d' ' -f 1").stdout.strip()
+        if published_checksum == local_checksum:
+            self.log(f"Checksums match. Download is OK.")
+            return True
+        elif published_checksum != local_checksum:
+            self.log(f"Checksums do not match. Download got corrupted.")
+            return False
+        else:
+            failure_exit(f"Error encountered when comparing checksums.")
+
+    def install_dependencies(self):
+        self.log("Installing dependencies")
+        self.exec("sudo apt-get update")
+        self.log(f"Install pv to show progress of extraction")
+        self.exec("sudo apt-get install -y pv")
+
+    def download_snapshot(self):
+        archive_name = os.path.basename(self.interviewer.snapshot_url)
+        self.mkdir_p(self.cheqd_data_dir)
+        # Fetch size of snapshot archive. Uses curl to fetch headers and looks for Content-Length.
+        archive_size = self.exec(f"curl --head {self.interviewer.snapshot_url} | awk  '/Length/ {{print $2}}'").stdout.strip()
+        # Check how much free disk space is available wherever the cheqd root directory is mounted
+        free_disk_space = self.exec("df -P {self.cheqd_root_dir} | tail -1 | cut -d' ' -f 3").stdout.strip()
+        if int(archive_size) < int(free_disk_space):
+            self.log(f"Downloading snapshot archive. This may take a while...")
+            self.exec(f"wget -c {self.interviewer.snapshot_url} -P {self.cheqd_root_dir}")
+            archive_path = os.path.join(self.cheqd_root_dir, archive_name)
+            if self.compare_checksum(archive_path) is True:
+                self.log(f"Snapshot download was successful and checksums match.")
+            else:
+                self.log(f"Snapshot download was successful but checksums do not match.")
+                failure_exit(f"Snapshot download was successful but checksums do not match.")
+        elif int(archive_size) > int(free_disk_space):
+            failure_exit (f"Snapshot archive is too large to fit in free disk space. Please free up some space and try again.")
+        else:
+            failure_exit (f"Error encountered when downloading snapshot archive.")
+        
+    def untar_from_snapshot(self):
+        archive_path = os.path.join(self.cheqd_root_dir, os.path.basename(self.interviewer.snapshot_url))
+        # Check if there is enough space to extract snapshot archive
+        free_disk_space = self.exec("df -P {self.cheqd_root_dir} | tail -1 | cut -d' ' -f 3").stdout.strip()
+        extracted_archive_estimate = self.exec(f"zcat {archive_path} | wc -c").stdout.strip()
```

Comment:
This instruction will extract the whole archive. It will take really long time and spend diskspace. I don't now other safe ways for getting info about extract space.